packages/dev/core/src/AudioV2/abstractAudio/subProperties/abstractSpatialAudio.ts

```diff
@@ -0,0 +1,188 @@
+import { Quaternion, Vector3 } from "../../../Maths/math.vector";
+
+export const _SpatialAudioDefaults = {
+    coneInnerAngle: 6.28318530718,
+    coneOuterAngle: 6.28318530718,
+    coneOuterVolume: 0,
+    distanceModel: "inverse" as DistanceModelType,
+    maxDistance: 10000,
+    panningModel: "equalpower" as PanningModelType,
+    position: Vector3.Zero(),
+    referenceDistance: 1,
+    rolloffFactor: 1,
+    rotation: Vector3.Zero(),
+    rotationQuaternion: Quaternion.FromEulerVector(Vector3.Zero()),
+};
+
+/**
+ * Options for spatial audio.
+ */
+export interface ISpatialAudioOptions {
+    /**
+     * The spatial cone inner angle, in radians. Defaults to 2π.
+     * - When the listener is inside the cone inner angle, the volume is at its maximum.
+     */
+    spatialConeInnerAngle: number;
+    /**
+     * The spatial cone outer angle, in radians. Defaults to 2π.
+     * - When the listener is between the the cone inner and outer angles, the volume fades to its minimum as the listener approaches the outer angle.
+     * - When the listener is outside the cone outer angle, the volume is at its minimum.
+     */
+    spatialConeOuterAngle: number;
+    /**
+     * The amount of volume reduction outside the {@link spatialConeOuterAngle}. Defaults to 0.
+     */
+    spatialConeOuterVolume: number;
+    /**
+     * The algorithm to use to reduce the volume of the audio source as it moves away from the listener. Defaults to "inverse".
+     * @see {@link spatialMaxDistance}
+     * @see {@link spatialReferenceDistance}
+     * @see {@link spatialRolloffFactor}
+     */
+    spatialDistanceModel: "linear" | "inverse" | "exponential";
```

Comment:
Same comment I had about the audio engine state - should be documented so JS devs could use it as well.

Reply:
Also note that the same is used in other parts of the framework, so be sure to document them as well.